### ISSUE TITLE
Bug: no-restricted-imports doesnt work.

### ISSUE THREAD

**Issue by Pentadome** [eslint/eslint#15043]
### Environment

Node version: v15.9.0
npm version: v7.21.1
Local ESLint version: v6.8.0 (Currently used)
Global ESLint version: Not found

### What parser are you using?

vue-eslint-parser

### What did you do?

```markdown
<details>
<summary>Configuration</summary>
"no-restricted-imports": ["error", {
    "patterns": [{
        "group": ["@vue/*"],
        "message": "Use `@vue` and not any sub-folders of '@vue'"
    }]
}]
</details>
```

from example at https://eslint.org/docs/rules/no-restricted-imports:
```
"no-restricted-imports": ["error", {
    "patterns": [{
      "group": ["import1/private/*"],
      "message": "usage of import1 private modules not allowed."
    }, {
      "group": ["import2/*", "!import2/good"],
      "message": "import2 is deprecated, except the modules in import2/good."
    }]
}]
```

### What did you expect to happen?

```markdown
An error when accidently typing `import { computed } from "@vue/runtime-core"`
instead of `import { computed } from "@vue"`
```

### What actually happened?

```markdown
Configuration for rule "no-restricted-imports" is invalid:
	Value {"patterns":[{"group":["@vue/*"],"message":"Use `@vue` and not any sub-folders of '@vue'"}]} should be string.
	Value {"patterns":[{"group":["@vue/*"],"message":"Use `@vue` and not any sub-folders of '@vue'"}]} should NOT have additional properties.
	Value {"patterns":[{"group":["@vue/*"],"message":"Use `@vue` and not any sub-folders of '@vue'"}]} should match some schema in anyOf.
	Value {"group":["@vue/*"],"message":"Use `@vue` and not any sub-folders of '@vue'"} should be string.
	Value [{"patterns":[{"group":["@vue/*"],"message":"Use `@vue` and not any sub-folders of '@vue'"}]}] should match some schema in anyOf.

  Code: -32603 
```
```


### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
Hi @Pentadome, thanks for the issue!

```json
"no-restricted-imports": ["error", {
    "patterns": [{
        "group": ["@vue/*"],
        "message": "Use `@vue` and not any sub-folders of '@vue'"
    }]
}]
```

This configuration for `no-restricted-imports` is supported only in ESLint 7.27.0 and newer versions, but you're using ESLint v6.8.0. You'll need to update to at least v7.27.0.

Here's an [online demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IFwibm8tcmVzdHJpY3RlZC1pbXBvcnRzXCI6IFtcImVycm9yXCIsIHtcbiAgICBcInBhdHRlcm5zXCI6IFt7XG4gICAgICAgIFwiZ3JvdXBcIjogW1wiQHZ1ZS8qXCJdLFxuICAgICAgICBcIm1lc3NhZ2VcIjogXCJVc2UgYEB2dWVgIGFuZCBub3QgYW55IHN1Yi1mb2xkZXJzIG9mICdAdnVlJ1wiXG4gICAgfV1cbn1dXG4qL1xuXG5pbXBvcnQgeyBjb21wdXRlZCB9IGZyb20gXCJAdnVlL3J1bnRpbWUtY29yZVwiOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTIsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=) (currently ESLint 7.32.0) with the same configuration.

**Comment by Pentadome** [comment]
Thanks!

Is it possible to see which configurations are allowed for 6.8.0? I can't upgrade yet unfortunately due to a dependency to a vue plugin for eslint.

**Comment by mdjermanovic** [comment]
Unfortunately, we don't have documentation for all ESLint versions on eslint.org, but you can see the 6.8.0 version on GitHub from v6.8.0 tag:

https://github.com/eslint/eslint/blob/v6.8.0/docs/rules/no-restricted-imports.md
